core/web/evm_transactions_controller.go

```diff
@@ -47,3 +60,131 @@ func (tc *TransactionsController) Show(c *gin.Context) {
 
 	jsonAPIResponse(c, presenters.NewEthTxResourceFromAttempt(*ethTxAttempt), "transaction")
 }
+
+type EvmTransactionController struct {
+	Enabled     bool
+	Logger      logger.SugaredLogger
+	TxmStorage  txmgr.EvmTxStore
+	AuditLogger audit.AuditLogger
+	Chains      evm.LegacyChainContainer
+	KeyStore    keystore.Eth
+}
+
+func NewEVMTransactionController(app chainlink.Application) *EvmTransactionController {
+	return &EvmTransactionController{
+		Enabled:     app.GetConfig().TxmAsService().Enabled(),
+		TxmStorage:  app.TxmStorageService(),
+		Logger:      app.GetLogger(),
+		AuditLogger: app.GetAuditLogger(),
+		Chains:      app.GetRelayers().LegacyEVMChains(),
+		KeyStore:    app.GetKeyStore().Eth(),
+	}
+}
+
+// Create signs and sends transaction from specified address. If address is not provided uses one of enabled keys for
+// specified chain.
+func (tc *EvmTransactionController) Create(c *gin.Context) {
+	if !tc.Enabled {
+		jsonAPIError(c, http.StatusUnprocessableEntity, errors.New("transactions creation disabled. To enable set TxmAsService.Enabled=true"))
+		return
+	}
+	var tx models.CreateEVMTransactionRequest
+	if err := c.ShouldBindJSON(&tx); err != nil {
+		jsonAPIError(c, http.StatusBadRequest, err)
+		return
+	}
+
+	if tx.IdempotencyKey == "" {
+		jsonAPIError(c, http.StatusBadRequest, errors.New("idempotencyKey must be set"))
+		return
+	}
+
+	decoded, err := hexutil.Decode(tx.EncodedPayload)
+	if err != nil {
+		jsonAPIError(c, http.StatusBadRequest, errors.Errorf("encodedPayload is malformed: %v", err))
+		return
+	}
+
+	if tx.ChainID == nil {
+		jsonAPIError(c, http.StatusBadRequest, errors.New("chainID must be set"))
+		return
+	}
+
+	if tx.ToAddress == nil {
+		jsonAPIError(c, http.StatusBadRequest, errors.New("toAddress must be set"))
+		return
+	}
+
+	chain, err := getChain(tc.Chains, tx.ChainID.String())
+	if err != nil {
+		if errors.Is(err, ErrMissingChainID) {
+			jsonAPIError(c, http.StatusUnprocessableEntity, err)
+			return
+		}
+
+		tc.Logger.Errorf("Failed to get chain", "err", err)
+		jsonAPIError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	if tx.FromAddress == utils.ZeroAddress {
+		tx.FromAddress, err = tc.KeyStore.GetRoundRobinAddress(tx.ChainID.ToInt())
```

Comment:
For now this is ok.
But I want to add a new feature, where the caller of TXM gives no fromAddress. And then TXM can choose the best fromAddress on its own.